main.tf

```diff
@@ -45,15 +23,14 @@ data "aws_subnet" "pod" {
 }
 
 locals {
-  # Get zones where ALL instance types are offered(intersection).
-  zone_intersection_instance_offerings = setintersection([for k, v in data.aws_ec2_instance_type_offerings.nodes : toset(v.locations)]...)
-  # Get the zones that are available and offered in the region for the instance types.
-  az_names    = var.vpc_id != null ? distinct(data.aws_subnet.private[*].availability_zone) : length(var.availability_zones) > 0 ? var.availability_zones : data.aws_availability_zones.available.names
-  offered_azs = setintersection(local.zone_intersection_instance_offerings, toset(local.az_names))
-  num_of_azs  = var.vpc_id != null ? length(local.az_names) : var.number_of_azs
+  zones_by_id = {
```

Comment:
why not just zip between names and zone_ids? They are in matching order.

Reply:
Goal was to provide a short-hand for the top-level module so people don't need to know the region identifier for each AZ. I'm not strongly for this, it's just repetitive.

Reply:
Pushed ff97d6762c554b843e019dcba2ea9b42807c7bd2, we can require folks to provide them explicitly.